main.py

```diff
@@ -1,33 +1,68 @@
-from telegram.ext import Application, CommandHandler, ContextTypes
+from telegram.ext import Application, CommandHandler, MessageHandler, ContextTypes, filters
+from commands import Commands
+import re
 
-# Load the .env file to read environment variables
-load_dotenv()
+class TelegramBot:
+    def __init__(self, token):
+        self.application = Application.builder().token(token).build()
+        self._add_command_handlers()
 
-# Get the Telegram bot token from the environment variable
-telegram_token = os.getenv('TELEGRAM_TOKEN')
+    def _add_command_handlers(self):
+        self.application.add_handler(CommandHandler(Commands.START.value, self.start))
+        self.application.add_handler(CommandHandler(Commands.CALCULATE.value, self.calculate))
+        self.application.add_handler(MessageHandler(filters.TEXT & ~filters.COMMAND, self.handle_message))
 
-# Define a command handler for the '/start' command
-async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
-    user = update.effective_user
-    await context.bot.send_message(
-        chat_id=update.effective_chat.id,
-        text=f"Hello, {user.first_name}! Welcome to our Telegram bot."
-    )
+    async def start(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
+        user = update.effective_user
+        await context.bot.send_message(
+            chat_id=update.effective_chat.id,
+            text=f"Hello, {user.first_name}! Welcome to our Telegram bot."
+        )
+        context.user_data["has_started"] = True
 
-# Define the main function to run the Telegram bot
-def main():
-    # Initialize the Telegram bot with the token
-    application = Application.builder().token(telegram_token).build()
+    async def calculate(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
+        if context.user_data.get("has_started"):
+            text = update.message.text
+            print(text)
+            keyword = "calculate"
+            keyword_pos = text.find(keyword)
+            expression = text[keyword_pos + len(keyword):].strip()
+            print(context.args, expression)
+            result = self.calculate_expression(expression)
+        else:
+            result="Unprocessable context, read comments at [some_link]"
 
-    # Add a handler for the '/start' command
-    application.add_handler(CommandHandler('start', start))
+        await context.bot.send_message(
+            chat_id=update.effective_chat.id,
+            text=result
+        )
 
-    # Start the bot to poll for messages
-    application.run_polling()
+    def calculate_expression(self, expression: str) -> str:
+        expression = expression.replace("plus", "+").replace("minus", "-")
+        valid_expression = "".join(re.findall(r"[\d]+|[-+]", expression))
+        try:
+            result = eval(valid_expression)
+            return f"Calculated with result: {result}"
+        except Exception:
+            return "Unprocessable context, read comments at [some_link]"
```

Comment:
idea was in some global link, which will be passed to a f-string, so user can take a look of how calculation works, so it should be in some related field, like Class attrs.